src/shell/autoexec.cpp

```diff
@@ -0,0 +1,584 @@
+/*
+ *  Copyright (C) 2020-2023  The DOSBox Staging Team
+ *  Copyright (C) 2002-2021  The DOSBox Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#include "autoexec.h"
+
+#include "checks.h"
+#include "control.h"
+#include "dosbox.h"
+#include "fs_utils.h"
+#include "setup.h"
+#include "shell.h"
+#include "string_utils.h"
+
+#include <algorithm>
+#include <iostream>
+#include <sstream>
+
+CHECK_NARROWING();
```

Comment:
This state machine processor is very nice - to finally have one collection and management point to sort through all the `autoexecs` collected from the (many) possible conf files.